### ISSUE TITLE
Overhaul of existing home screen

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6361]
While there are some existing issues to redesign the complete app UI or major parts of it (and still we have no one around capable of doing this) might I suggest to at least to improve the existing UI a little.

Most devices now have a high resolution screen but our main menu is still optimized for low resolution it seems.

I mean:
When looking at the main menu I do see a lot of empty space and six items in the middle.
We could at least place the icons more distinct from each other and modify the text boxes to show the complete text (right now it shows for me e.g. "Gespeich..." where there is enough space for the complete title).

Here is my screen with almost all items descriptions shortened:
![screenshot_20170226-222303](https://cloud.githubusercontent.com/assets/949669/23343870/ef37e60e-fc72-11e6-93b7-d49f54ecdbec.png)

**Comment by pstorch** [comment]
What about starting with the Map right away? 
Let the user choose his preferred list of caches to display on the Map (for me it would be my homezone list) or use the Live Map if nothing is configured yet.

**Comment by SammysHP** [comment]
Not sure about that. I don't use the map that often.

**Comment by Bananeweizen** [comment]
I would also not jump to the map immediately, especially since the map requires either online connection or stored map data to be useful. Nevertheless I'd like to see more actionable content instead of just 6 links leading to content afterwards.

**Comment by kumy** [comment]
Personnaly I prefer the list activity. Why not making it (choosing a strtup activity) an optionnal configurable parameter?

**Comment by Lineflyer** [comment]
I don't think it is meaningful just to jump to any function directly.
The intention of this issue I opened was just to ask if we could give the existing icons more space on the screen but not about anything else.
There are other issues concerning a complete redesign or adding new buttons. This small task described here would just give a quick win which hopefully is not so difficult to realize.

BTW: AFAIR we already have widgets allowing the user to jump directly into some functions (although they might be rarely known).

**Comment by dcz-switcher** [comment]
Hi !
I'm a new c:geo user
I'm agree with this issue and here is my proposal

note: do you have statistics of the app usage ?
If 90% of people tap on map (for example) when the app starts, it maybe interesting to start with this activity first, no ? but as Lineflyer stated, this is not the purpose of this issue ;-)

![home screen](https://cloud.githubusercontent.com/assets/299807/26512819/aba78408-4268-11e7-88b4-186c1199e1c6.png)

**Comment by dcz-switcher** [comment]
... and because the actual theme is dark, here is the same in dark mode
I'm not a big fan of dark interfaces except for my IDE ^_^
but make small changes is more realistic

![home screen black](https://cloud.githubusercontent.com/assets/299807/26513949/d02ce9f2-426e-11e7-8a89-206de07d41f8.png)

**Comment by SammysHP** [comment]
But now a lot of important information is missing and there's no benefit for the user because it has the same six entries.

**Comment by dcz-switcher** [comment]
It's true, so here is a new version with all (I hope) informations
I choose to put the nav icons at the bottom of the screen because they are easier to reach with a thumb
On top, I put the *.caching accounts
In the middle, the location informations

> there's no benefit for the user because it has the same six entries
the purpose of this issue is not change the six entries, but use all space available of the device and not truncate the text

What do you think of this new version ?

![home screen complet](https://cloud.githubusercontent.com/assets/299807/26528391/41cabb24-43ab-11e7-9d03-2cd5600c318e.png)

**Comment by moving-bits** [comment]
Regarding the original proposal by @Lineflyer

> We could at least place the icons more distinct from each other and modify the text boxes to show the complete text (right now it shows for me e.g. "Gespeich..." where there is enough space for the complete title).

I did some tweaking to the mainscreen's layout configuration, so that icons and text are evenly distributed in three columns. Works out fine on higher resolutions as well as on small screens, both tested with english and german language settings:

![2019-01-03 mainscreen 3 column german](https://user-images.githubusercontent.com/3754370/50655398-535fb800-0f90-11e9-974a-52f9ab84f684.png)

![2019-01-03 mainscreen 3 column english](https://user-images.githubusercontent.com/3754370/50655406-56f33f00-0f90-11e9-802b-653fce291b5d.png)

This fixes the problem with cut off text (as long as there is enough space), we could close this issue, and use 7030 and others for future UI discussion. What do you think?

**Comment by Lineflyer** [comment]
My opinion: As initially emphasized its the only feasible way to improve in small steps. And this could be a first step, which might (hopefully) not last too long until we have something even better. But I like it as a quick improvement.

**Comment by moving-bits** [comment]
The rows are self-aligning horizontally, but text is fixed to one line currently. I'll look into it.
Maybe we need to reduce the top margin a little bit to gain some space for another line. But: all translations for those six texts have to be short enough to not become three or even more lines, otherwise it will overflow on small screens.

**Comment by moving-bits** [comment]
Found an solution: I can limit it to "maxLines=2" - if every text in a row is a one liner, then there's no difference, otherwise that line gets more space automatically. If the text exceeds two lines, it gets truncated (sorry, without an indicator - but that's ok, IMHO). Works on large devices as well as on smaller ones.

![2019-01-03 mainscreen 3 column max 2 lines english large device](https://user-images.githubusercontent.com/3754370/50658301-324f9500-0f99-11e9-8c6d-cd2bf44b22e2.png)

![2019-01-03 mainscreen 3 column max 2 lines english](https://user-images.githubusercontent.com/3754370/50658304-35e31c00-0f99-11e9-94d8-237180455a0d.png)

I'll create an PR.

**Comment by bekuno** [comment]
Maybe we need a separate configuration for the landscape mode. (only on line text, but more space left and right)

**Comment by moving-bits** [comment]
generally it works in landscape mode as well:

![2019-01-03 mainscreen 3 column max 2 lines english landscape](https://user-images.githubusercontent.com/3754370/50658548-08e33900-0f9a-11e9-90dc-741afd4de30c.png)

**Comment by bekuno** [comment]
Are the buttons in the lower line useable?
With more lines for further login data they would be hidden.

**Comment by moving-bits** [comment]
and screenshot for large screen device:

![image](https://user-images.githubusercontent.com/3754370/50658769-b3f3f280-0f9a-11e9-8495-5e00a97c5271.png)

**Comment by moving-bits** [comment]
> 
> 
> Are the buttons in the lower line useable?
> With more lines for further login data they would be hidden.

yes, as long as the buttons are not completely covered by the overlays

**Comment by bekuno** [comment]
Its possible to reduce the top space or (re)calculate the place with under consideration of this lines?
(you could have eg. more then one OC account, EC, SU (in the next time))

**Comment by moving-bits** [comment]
> Its possible to reduce the top space or (re)calculate the place with under consideration of this lines?
> (you could have eg. more then one OC account, EC, SU (in the next time))

don't know how to, as all those layout things are done in xml files

**Comment by Eldenroot** [comment]
First step :)

another will be a GUI overhaul :)

I like this:
https://cloud.githubusercontent.com/assets/299807/26512819/aba78408-4268-11e7-88b4-186c1199e1c6.png

**Comment by SammysHP** [comment]
Should we reopen this issue for the other ideas and further improvements?

**Comment by Lineflyer** [comment]
We also still have #7030
